core/base/src/TROOT.cxx

```diff
@@ -1558,19 +1558,19 @@ TStyle *TROOT::GetStyle(const char *name) const
 
 TObject *TROOT::GetFunction(const char *name) const
 {
-   if (name == nullptr || name[0] == 0) {
+   if (!name || !*name)
       return nullptr;
-   }
 
-   {
-      R__LOCKGUARD(gROOTMutex);
-      TObject *f1 = fFunctions->FindObject(name);
-      if (f1) return f1;
-   }
+   static std::atomic<bool> isInited = false;
+   bool wasInited = isInited.load();
```

Comment:
```suggestion
   // Capture the state before calling FindObject as it could change
   // between the end of FindObject and the if statement.
   bool wasInited = isInited.load();
```